addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/HttpsTrustManager.java

```
@@ -0,0 +1,85 @@
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * The {@link HttpsTrustManager} is responsible for ignoring invalid HTTPS certificates.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+
+public class HttpsTrustManager implements X509TrustManager {
+
+    private static TrustManager[] trustManagers;
+    private static final X509Certificate[] ACCEPTED_ISSUERS = new X509Certificate[] {};
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] x509Certificates, String s)
+            throws java.security.cert.CertificateException {
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] x509Certificates, String s)
+            throws java.security.cert.CertificateException {
+    }
+
+    public boolean isClientTrusted(X509Certificate[] chain) {
+        return true;
+    }
+
+    public boolean isServerTrusted(X509Certificate[] chain) {
+        return true;
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+        return ACCEPTED_ISSUERS;
+    }
+
+    public static void allowAllSSL() {
+        HttpsURLConnection.setDefaultHostnameVerifier(new HostnameVerifier() {
+
+            @Override
+            public boolean verify(String arg0, SSLSession arg1) {
+                return true;
+            }
+
+        });
+
+        SSLContext context = null;
+        if (trustManagers == null) {
+            trustManagers = new TrustManager[] { new HttpsTrustManager() };
+        }
+
+        try {
+            context = SSLContext.getInstance("TLS");
+            context.init(null, trustManagers, new SecureRandom());
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+            // e.printStackTrace();
```

Do not ignore exceptions